riscv/src/register/mvendorid.rs

```diff
@@ -1,34 +1,8 @@
 //! mvendorid register
 
-use core::num::NonZeroUsize;
-
-/// mvendorid register
-#[derive(Clone, Copy, Debug)]
-pub struct Mvendorid {
-    bits: NonZeroUsize,
-}
-
-impl Mvendorid {
-    /// Returns the contents of the register as raw bits
-    #[inline]
-    pub fn bits(&self) -> usize {
-        self.bits.get()
-    }
-
-    /// Returns the JEDEC manufacturer ID
-    #[inline]
-    pub fn jedec_manufacturer(&self) -> usize {
```

Comment:
Should we add the `jedec_manufacturer` field?

Reply:
Good question, I missed that function when converting to the CSR macro.

After reading the ISA again, it looks like the `jedec_manufacturer` implementation was incorrect.


```
mvendorid encodes the number of one-byte continuation codes in the Bank field, and
encodes the final byte in the Offset field, discarding the parity bit. For example, the JEDEC
manufacturer ID 0x7f 0x7f 0x7f 0x7f 0x7f 0x7f 0x7f 0x7f 0x7f 0x7f 0x7f 0x7f 0x8a (twelve
continuation codes followed by 0x8a) would be encoded in the mvendorid CSR as 0x60a.
```

So, we would probably want to add `bank` and `offset` fields, along with a correct `jedec_manufacturer` function that decodes the ID:

```rust
pub const fn jedec_manufacturer(&self) -> (usize, usize) {
    (self.bank(), 0x80 | self.offset())
}
```